biapol_utilities/label/_match_labels.py

```diff
@@ -61,6 +62,10 @@ def match_labels(label_image_x, label_image_y, method=intersection_over_union, *
     if method == intersection_over_union:
         threshold = kwargs.get('iou_threshold', 0.25)
     
+    # relabel label_image_y to keep overlap matrix small
+    label_image_y, _, _ = relabel_sequential(label_image_y)
```

Comment:
Great solution. I fully agree that here in this context relabeling makes sense. :-)

Reply:
Thanks! :)